controller/api/destination/endpoint_profile_translator.go

```diff
@@ -55,25 +55,6 @@ func (ept *endpointProfileTranslator) Update(address *watcher.Address) (bool, er
 		return false, fmt.Errorf("failed to create endpoint: %w", err)
 	}
 
-	// The protocol for an endpoint should only be updated if there is a pod,
```

Comment:
All of this protocol hinting logic is already implemented in `createWeightedAddr` and the endpoint that we get from `createEndpoint` already has the protocol hinting metadata set on it correctly.  This code block is redundant.

Reply:
Is this function unit tested? Shouldn't a test break somewhere if so?